test/producer.test.js

```diff
@@ -324,42 +326,90 @@ describe('producer', () => {
       });
     }
 
-    before('stop nsqd', done => {
-      spawn('docker-compose', [`--file=${composeFile}`, 'stop', 'nsqd3']).on('close', done);
+    it('should raise error when nsqd is down and maximum retries are exhuasted', done => {
+      const p = new Producer({
+        nsqdHost: 'localhost',
+        tcpPort: 9040
+      });
+      const errMsg = 'Failed to connect';
+      const retryOpt = {
+        retries: 3,
+        minTimeout: 300
+      };
+
+      let retryTimes = 0;
+      p.connect = () => {
```

Comment:
I think you are altering the interface of the function `connect`, it was a function that accepts a node-style callback instead of returning a Promise.

Reply:
> I think you are altering the interface of the function `connect`, it was a function that accepts a node-style callback instead of returning a Promise.

No, the connect method could be used either as a traditional nodejs callback style, or could be used as a promise style. According to the api document here for the asCallback method: http://bluebirdjs.com/docs/api/ascallback.html, it simply will make everyone happy.